lib/services/elasticsearch.js

```diff
@@ -153,665 +173,682 @@ class ElasticSearch extends Service {
-
-    esRequest.id = request.input.resource._id;
+  delete (index, collection, id, { refresh=false } = {}) {
+    const esRequest = {
+      index: this._buildClient._getESIndex(index, collection),
+      id,
+      refresh,
+      retryOnConflict
+    };
 
     assertWellFormedRefresh(esRequest);
 
-    return this.client
-      .get({
-        index: esRequest.index,
-        type: esRequest.type,
-        id: esRequest.id
-      })
-      .then(result => {
-        if (result._source._kuzzle_info && !result._source._kuzzle_info.active) {
-          this.throw('document_not_found', esRequest.id);
-        }
-        // injecting retryOnConflict default configuration
-        if (
-          !_.has(esRequest, 'retryOnConflict')
-          && this.config.defaults.onUpdateConflictRetries > 0
-        ) {
-          esRequest.retryOnConflict = this.config.defaults.onUpdateConflictRetries;
-        }
-
-        // Add metadata
-        esRequest.body = {
-          doc: {
-            _kuzzle_info: {
-              active: false,
-              deletedAt: Date.now(),
-              updater: this._getUserId(request)
-            }
-          }
+    return this.client.delete(esRequest)
+      .then(() => {
+        const apiResponse = {
+          _id: id
         };
 
-        return this.client.update(esRequest)
-          .then(r => this.refreshIndexIfNeeded(esRequest, r))
-          .catch(error => Bluebird.reject(
-            this.esWrapper.formatESError(error))
-          );
+        return apiResponse;
       })
-      .catch(error => Bluebird.reject(
-        this.esWrapper.formatESError(error))
-      );
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
    * Delete all documents matching the provided filters
    *
-   * @param {Request} request
-   * @returns {Promise} resolve an object with ids
+   * @param {string} index - Index name
+   * @param {string} collection - Collection name
+   * @param {object} query - Query to match documents
+   * @param {object} options - from (null), size (null), refresh (false)
+   *
+   * @returns {Promise} { total, deleted, failures }
    */
-  deleteByQuery(request) {
-    const
-      esRequestSearch = initESRequest(request, ['from', 'size', 'scroll']),
-      esRequestBulk = initESRequest(request, ['refresh']),
-      ts = Date.now();
+  deleteByQuery (
+    index,
+    collection,
+    query,
+    { refresh=false, from=null, size=null } = {})
+  {
+    const esRequest = {
+      index: this._getESIndex(index, collection),
+      body: { query },
+      scroll: '30s',
+      from,
+      size,
+      refresh
+    };
 
-    assertWellFormedRefresh(esRequestBulk);
+    if (! _.isPlainObject(query)) {
+      return this.reject('empty_query');
+    }
 
-    esRequestSearch.body = request.input.body;
-    esRequestSearch.scroll = '30s';
-    esRequestBulk.body = [];
+    return this.client.deleteByQuery(esRequest)
+      .then(({ body }) => {
+        const apiResponse = {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "apiResponse". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1488)